documentation/Amazon-Managed-Grafana/ReadMe.md

```diff
@@ -0,0 +1,149 @@
+kind: ExternalSecret
+metadata:
+  name: grafana-token-secret
+  namespace: grafana
+spec:
+  refreshInterval: 1m
+  secretStoreRef:
+    name: grafana-secret-store #The secret store name we have just created.
+    kind: ClusterSecretStore
+  target:
+    name: grafana-secret # Secret name in k8s
+  data:
+  - secretKey: grafana-token # which key it's going to be stored
+    remoteRef:
+      key: grafana-api-key # Our secret-name goes here
+```
+
+### Adding AWS data sources to Amazon Managed Grafana
+
+Today Grafana operator supports [Amazon Managed Service for Prometheus](https://github.com/grafana-operator/grafana-operator/blob/master/deploy/examples/datasources/AWS-Prometheus.yaml) as a data source. Grafana operator solution should be enhanced to add following AWS data sources (minimally Amazon CloudWatch) :
+- Amazon CloudWatch
+- Amazon OpenSearch Service
+- AWS IoT SiteWise
+- AWS IoT TwinMaker
+- Amazon Timestream
+- Amazon Athena
+- Amazon Redshift
+- AWS X-Ray
+
+This is not a road block feature but nice to have to support all AWS data sources.
+
+In order to accomplish this, "Reading keys for accessing Amazon Managed Grafana from AWS Secrets Manager" feature above becomes a prerequisite. `GrafanaDataSource` CRD should be modified to use `grafana-secret` to add AWS data sources as `GrafanaDataSource` on Amazon Managed Grafana. This allows customers to use GitOps approach to add AWS data sources as `GrafanaDataSource` on remote Grafana instances such as Amazon Managed Grafana.
+
+### Creating Grafana dashboards on Amazon Managed Grafana.
+
+Today Grafana operator supports the creation of Grafana dashboards on self managed grafana instance in the cluster where the Grafana operator is installed. Grafana operator should be enhanced to support creating dashboards on Amazon Managed Grafana. 
+
+In order to accomplish this, "Reading keys for accessing Amazon Managed Grafana from AWS Secrets Manager" feature above becomes a prerequisite. `GrafanaDashboard` CRD should be modified to use `grafana-secret` to create `GrafanaDashboard` on Amazon Managed Grafana. This allows customers to use GitOps approach to create Grafana Dashboards on remote Grafana instances such as Amazon Managed Grafana.
+
+### Creating Grafana alerting on Amazon Managed Grafana.
+
+Today Grafana operator supports the creation of Grafana notification on self managed grafana instance in the cluster where Grafana operator is installed. Grafana operator should be enhanced to support creating notifications on Amazon Managed Grafana. 
+
+In order to accomplish this, "Reading keys for accessing Amazon Managed Grafana from AWS Secrets Manager" feature above becomes a prerequisite. `GrafanaNotificationChannel` CRD should be modified to use `grafana-secret` to create `GrafanaNotificationChannel` on Amazon Managed Grafana. This allows customers to use GitOps approach to create Grafana alerting on remote Grafana instances such as Amazon Managed Grafana.
```

Comment:
NotificationChannel is out of support in grafana 9 and the new version of the operator we will only support grafana 9 and forward.

Reply:
Sounds good, We need a mechanism to support alerting on external grafana instances.

Reply:
It's a separate CRD and it's out of scope for this design.
See: https://github.com/grafana-operator/grafana-operator-experimental/issues/38